### PR TITLE
raftstore: update follower size and keys after merge

### PR DESCRIPTION
**Pull request by Connor1996**
## What have you changed? (mandatory)
When commit merge, we only update approximate size and keys for leader, but there is a issue that after merge and then transfer leader, the new leader may have stale size and keys.

## What are the type of the changes? (mandatory)
- Bug fix (non-breaking change which fixes an issue)

## How has this PR been tested? (mandatory)
unit-test

## Does this PR affect documentation (docs/docs-cn) update? (mandatory)
No

## Does this PR affect tidb-ansible update? (mandatory)
No

### CODE REVIEW

File: tests/raftstore_cases/test_merge.rs [hunk 1]
```diff
@@ -449,41 +448,59 @@ fn test_merge_approximate_size_and_keys() {
...
+    // make sure split check is invoked so size and keys are updated.
     thread::sleep(Duration::from_secs(1));
 
     let left = pd_client.get_region(b"").unwrap();
     let right = pd_client.get_region(&max_key).unwrap();
-
     assert_ne!(left, right);
+
+    // make sure all peer's approximate size is not None.
+    cluster.must_transfer_leader(right.get_id(), right.get_peers()[0].clone());
```
Comment (siddontang) [inline tests/raftstore_cases/test_merge.rs]: em, you totally sleep too long for this test, can you reduce the sleep time?

File: src/raftstore/store/fsm/peer.rs [hunk 2]
```diff
@@ -1264,6 +1264,10 @@ impl<T: Transport, C: PdClient> Store<T, C> {
             peer.size_diff_hint = self.cfg.region_split_check_diff.0;
```
Comment (overvenus) [inline src/raftstore/store/fsm/peer.rs]: Should we update `size_diff_hint` on followers?

File: tests/raftstore_cases/test_merge.rs [hunk 3]
```diff
@@ -449,41 +448,59 @@ fn test_merge_approximate_size_and_keys() {
...
-            .get_region_approximate_keys(right.get_id())
-            .unwrap();
+        .get_region_approximate_keys(right.get_id())
+        .unwrap();
     assert_ne!(keys, 0);
 
     pd_client.must_merge(left.get_id(), right.get_id());
-    thread::sleep(Duration::from_secs(1));
+    // make sure split check is invoked so size and keys are updated.
+    thread::sleep(Duration::from_secs(2));
```
Comment (overvenus) [inline tests/raftstore_cases/test_merge.rs]: Why sleep guarantees that split check is invoked?

File: src/raftstore/store/fsm/peer.rs (hunk 2 above)
Comment (Connor1996) [reply src/raftstore/store/fsm/peer.rs]: update `size_diff_hint` or `approximate_size` does same effect, it is all about to invoke size check.

File: tests/raftstore_cases/test_merge.rs (hunk 3 above)
Comment (Connor1996) [reply tests/raftstore_cases/test_merge.rs]: split check every 100ms

**Comment by Connor1996** [comment]
@overvenus PTAL

File: tests/raftstore_cases/test_merge.rs (hunk 1 above)
Comment (overvenus) [reply tests/raftstore_cases/test_merge.rs]: How to guarantee approximate size is updated?

**Pull request merged**